client/src/pages/Messages/ChatBox.tsx

```diff
@@ -0,0 +1,187 @@
+import {
+  Avatar,
+  Backdrop,
+  Badge,
+  Box,
+  Button,
+  Divider,
+  IconButton,
+  LinearProgress,
+  Paper,
+  TextField,
+  Typography,
+} from '@material-ui/core';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
+import clsx from 'clsx';
+import { ChangeEvent, useEffect, useState } from 'react';
+import { useConversation } from '../../context/useConversationContext';
+import { getOneConversation } from '../../helpers/APICalls/conversations';
+import { sendMessage } from '../../helpers/APICalls/messages';
+import { Profile } from '../../interface/Profile';
+import { RecipientChatText } from './RecipientChatText';
+import { SenderChatText } from './SenderChatText';
+import useStyles from './useStyles';
+import { useSnackBar } from '../../context/useSnackbarContext';
+
+interface Props {
+  profileId: string;
+}
+
+const ChatBox = ({ profileId }: Props): JSX.Element => {
+  const { badgeStyle, chatBoxAvatar, chatBoxfullName, chatBox, sendButton, replyArea, backdrop } = useStyles();
+  const userBadgeStyle = true ? clsx(badgeStyle, 'active') : badgeStyle;
+  const { currentConversation, conversations, updateCurrentConversation } = useConversation();
+  const [currentRecipient, setCurrentRecipient] = useState<Profile | undefined>();
+  const [messageText, setMessageText] = useState('');
+  const { updateSnackBarMessage } = useSnackBar();
+
+  const handleSendMessage = () => {
+    const AsyncSendMessage = async () => {
+      if (currentConversation && currentRecipient && currentConversation.conversationId && currentRecipient._id) {
+        const conversationId = currentConversation.conversationId;
+        const recipientId = currentRecipient._id;
+
+        const response = await sendMessage(conversationId, messageText, recipientId);
+        if (response.error) {
+          updateSnackBarMessage(response.error);
+        } else if (response.message) {
+          const setCurrentConversation = async () => {
+            const response = await getOneConversation(conversationId);
+
+            if (response.error) {
+              updateSnackBarMessage(response.error);
+            } else if (response.conversationContent) {
+              updateCurrentConversation(response.conversationContent);
+            }
+          };
+
+          setCurrentConversation();
```

Comment:
You're already in an `async` function. So no need to declare this one, rather you could just go ahead and use the function body.
 `const response = await ...`
 Let me know if that's not clear.

Reply:
done ✔️